include/seqan3/core/type_traits/function.hpp

```diff
@@ -16,8 +16,115 @@
 
 #include <seqan3/core/type_list/traits.hpp>
 
+namespace seqan3::detail
+{
+// ----------------------------------------------------------------------------
+// multi_invocable
+// ----------------------------------------------------------------------------
+
+/*!\brief A type that can conveniently inherit multiple invocables and acts as a union over them.
+ * \tparam invocable_ts The types to inherit from.
+ * \ingroup type_traits
+ */
+template <typename ...invocable_ts>
+struct multi_invocable : invocable_ts...
+{
+    //!\brief Inherit the function call operators.
+    using invocable_ts::operator()...;
+};
+
+//!\brief Deduction guides for seqan3::detail::multi_invocable.
+template <typename ...invocable_ts>
+multi_invocable(invocable_ts...) -> multi_invocable<invocable_ts...>;
```

Comment:
This change is noise and I would have preferred if old things stay at old places (extremely minimizes merge conflict potential). If you want to make this change, we can do it in a different PR.